2996_reflection/reflection.md

```diff
@@ -3576,12 +3550,22 @@ Add a new paragraph between [expr.eq]{.sref}/5 and /6:
 
 ### [expr.const]{.sref} Constant Expressions {-}
 
-Add a new paragraph after the definition of _manifestly constant-evaluated_ [expr.const]{.sref}/20:
+Add a new paragraph after the definition of _potentially constant-evaluated_ [expr.const]{.sref}/21:
+
+::: std
+::: addu
+
+[22]{.pnum} The _evaluation context_ is a set of points within the program that determines which declarations are visible to certain expressions used for reflection. During the evaluation of a set of manifestly constant-evaluated expressions `$M$`, let `$E$` be the unique such expression whose result is required for the evaluation of every other expression in `$M$`. The evaluation context comprises the points in the instantiation context of `$E$` ([module.context]).
```

Comment:
I don't know if we need to say this. Following Gaby's feedback, I'm thinking all reachable declarations are presumably usable via reflection ... so this concept of scoping the reflection to what's reachable from the translation unit is perhaps, implicit? Or maybe I don't know exactly what you're going for here?

Reply:
We talked a bit more about this offline - the trouble is that you cannot say that a declaration (member) is simply "reachable", you have to specify where it's reachable _from_ (which is defined using a "program point"). That boils down to two cases:

1. The declaration _D_ is in the same TU as _P_. In that case, it's reachable if _D_ precedes _P_.

2. The declaration _D_ is in a different TU than _P_. Note that the set of "necessarily reachable TUs" is _also_ defined in terms of a program point - so for instance, the declarations from an imported module are not visible prior to the associated `import`-statement (https://eel.is/c++draft/module.reach#1).

In either case, you cannot (formally) discuss reachability without specifying a point - so this is my attempt to formalize the notion that Daveed has called the "point of evaluation".